QQE-1584.md

```diff
@@ -0,0 +1,82 @@
+# QQE-1584 Quarkus LangChain4J testing
+
+JIRA link: https://issues.redhat.com/browse/QQE-1584
+
+This is a high-level strategy for testing of langchain4j
+
+## Scope of the testing
+@SystemMessage + @UserMessage annotations
+RAG functionality
+Integration with MCP API
+
+### Optional, but desirable:
+@Tool + @RegisterAiService (only supported for OpenAI, so cannot be tested locally)
+Observability for tool requests (to avoid a situation, where LLM overwhelms the database with too many requests)
+
+## Existing coverage upstream:
+### Workshop
+https://quarkus.io/quarkus-workshop-langchain4j/step-11/#running-the-llm-inference-locally  and  https://github.com/quarkusio/quarkus-workshop-langchain4j/tree/main/step-11
+(they are not run automatically, as far as I can tell)
+The tests use jlama, which requires Java 21+
```

Comment:
Part of the strategy should be figuring out how to get this to be runnable with product supported platforms.

I presume if we work with tech preview, just the main version of JDK should be enough, but it still should be considered.

Reply:
It should be written somewhere :)